src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/respondentsreport/RespondentsReportTest.java

```diff
@@ -0,0 +1,113 @@
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.ecm.common.model.helper.Constants.*;
+import uk.gov.hmcts.ecm.common.model.reports.respondentsreport.RespondentsReportSubmitEvent;
+
+public class RespondentsReportTest {
+
+    RespondentsReportDataSource reportDataSource;
+    RespondentsReport respondentsReport;
+    RespondentsReportCaseDataBuilder caseDataBuilder;
+    List<RespondentsReportSubmitEvent> submitEvents = new ArrayList<>();
+    static final LocalDateTime BASE_DATE = LocalDateTime.of(2022, 1, 1, 0, 0,0);
+    static final String DATE_FROM = BASE_DATE.minusDays(1).format(OLD_DATE_TIME_PATTERN);
+    static final String DATE_TO = BASE_DATE.plusDays(15).format(OLD_DATE_TIME_PATTERN);
+    public RespondentsReportTest() {
+        caseDataBuilder = new RespondentsReportCaseDataBuilder();
+    }
+
+    @Before
+    public void setup() {
+        submitEvents.clear();
+        reportDataSource = mock(RespondentsReportDataSource.class);
+        when(reportDataSource.getData(MANCHESTER_CASE_TYPE_ID, DATE_FROM, DATE_TO)).thenReturn(submitEvents);
+        respondentsReport = new RespondentsReport(reportDataSource,DATE_FROM, DATE_TO);
+    }
+
+    @Test
+    public void shouldNotShowCaseWithNoRespondent() {
+        // Given a case has no respondent
+        // and report data is requested
+        // the case should not be in the report data
+
+            submitEvents.clear();
+            caseDataBuilder = new RespondentsReportCaseDataBuilder();
+            caseDataBuilder.withNoRespondents();
+            submitEvents.add(caseDataBuilder
+                    .buildAsSubmitEvent());
+
+            var reportData = respondentsReport.generateReport(MANCHESTER_LISTING_CASE_TYPE_ID);
+            assertCommonValues(reportData);
+            assertEquals("0", reportData.getReportSummary().getTotalCasesWithMoreThanOneRespondent());
+            assertEquals(0, reportData.getReportDetails().size());
+    }
+
+    @Test
+    public void shouldNotShowCaseWithOneRespondent() {
+        // Given a case has 1 respondent
+        // and report data is requested
+        // the case should not be in the report data
+
+        caseDataBuilder.withOneRespondent();
+        submitEvents.add(caseDataBuilder
+                .buildAsSubmitEvent());
+
+        var reportData = respondentsReport.generateReport(MANCHESTER_LISTING_CASE_TYPE_ID);
+        assertCommonValues(reportData);
+        assertEquals("0", reportData.getReportSummary().getTotalCasesWithMoreThanOneRespondent());
+        assertTrue(reportData.getReportDetails().isEmpty());
+    }
+
+    @Test
+    public void shouldShowCaseWithMoreThanOneRespondent() {
+        // Given a case has more than 1 respondents
+        // and report data is requested
+        // the cases should be in the report data
+
+        caseDataBuilder.withMoreThanOneRespondents();
+        submitEvents.add(caseDataBuilder
+                .buildAsSubmitEvent());
```

Comment:
I think this would fit on the line above

Reply:
yeah done